src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -572,8 +572,7 @@ private void WriteMembers(Member[] members, UnknownNodeAction elementElseAction,
                 }
                 else
                 {
-                    Type elementType = collectionType.GetElementType()!;
-                    a = Array.CreateInstance(elementType, collectionMember.Count);
+                    a = Array.CreateInstanceFromArrayType(collectionType, collectionMember.Count);
```

Comment:
You can also fix the TODO here: Replace with https://github.com/dotnet/runtime/issues/76478

It should be fixed by creating the array using `Array.CreateInstanceFromArrayType` and then calling `ArrayList.CopyTo` to fill it in.